lib/ansible/modules/cloud/vmware/vmware_vswitch.py

```diff
@@ -195,179 +359,424 @@ def process_state(self):
+        vss_spec.policy.nicTeaming.nicOrder.activeNic = self.teaming_failover_order_active
+        vss_spec.policy.nicTeaming.nicOrder.standbyNic = self.teaming_failover_order_standby
+        vss_spec.policy.nicTeaming.failureCriteria = vim.host.NetworkPolicy.NicFailureCriteria()
+        if self.teaming_failure_detection == "link_status_only":
+            vss_spec.policy.nicTeaming.failureCriteria.checkBeacon = False
+        elif self.teaming_failure_detection == "beacon_probing":
+            vss_spec.policy.nicTeaming.failureCriteria.checkBeacon = True
+        # The following properties are deprecated since VI API 5.1. Default values are used
+        vss_spec.policy.nicTeaming.failureCriteria.fullDuplex = False
+        vss_spec.policy.nicTeaming.failureCriteria.percentage = 0
+        vss_spec.policy.nicTeaming.failureCriteria.checkErrorPercent = False
+        vss_spec.policy.nicTeaming.failureCriteria.checkDuplex = False
+        vss_spec.policy.nicTeaming.failureCriteria.speed = 10
+        vss_spec.policy.nicTeaming.failureCriteria.checkSpeed = 'minimum'
+        if self.module.check_mode:
+            results['changed'] = True
+            results['result'] = "vSwitch '%s' will be created" % self.switch
+        else:
+            try:
+                network_mgr = self.host_system.configManager.networkSystem
+                if network_mgr:
+                    network_mgr.AddVirtualSwitch(vswitchName=self.switch,
+                                                 spec=vss_spec)
+                    results['changed'] = True
+                    results['result'] = "vSwitch '%s' was created successfully" % self.switch
+                else:
+                    self.module.fail_json(msg="Failed to find network manager for ESXi system")
+            except vim.fault.AlreadyExists as already_exists:
+                results['result'] = "vSwitch with name %s already exists: %s" % (self.switch,
+                                                                                 to_native(already_exists.msg))
+            except vim.fault.ResourceInUse as resource_used:
+                self.module.fail_json(msg="Failed to add vSwitch '%s' as physical network adapter"
+                                          " being bridged is already in use: %s" % (self.switch,
+                                                                                    to_native(resource_used.msg)))
+            except vim.fault.HostConfigFault as host_config_fault:
+                self.module.fail_json(msg="Failed to add vSwitch '%s' due to host"
+                                          " configuration fault : %s" % (self.switch,
+                                                                         to_native(host_config_fault.msg)))
+            except vmodl.fault.InvalidArgument as invalid_argument:
+                self.module.fail_json(msg="Failed to add vSwitch '%s', this can be due to either of following :"
```

Comment:
This will concatenate all chunks into a single-line string. Try multiline:
```suggestion
                self.module.fail_json(msg=textwrap.dedent("""
                    Failed to add vSwitch '%s', this can be due to either of following:
                     1. vSwitch Name exceeds the maximum allowed length,
                     2. Number of ports specified falls out of valid range,
                     3. Network policy is invalid,
                     4. Beacon configuration is invalid : %s
                    """) % (self.switch, to_native(invalid_argument.msg)))
```